### DISCUSSION TITLE
Create a URL w/o header?

### DISCUSSION THREAD

**Discussion by bc3tech** [shlinkio/shlink#1974] [Help wanted] [answered]
### Shlink version

3.7.3

### PHP version

8.3

### How do you serve Shlink

Self-hosted Apache

### Database engine

MySQL

### Database version

5.7.23-23

### Summary

I found issue #147 but don't think it's been implemented based on following the trail. However what I am wondering is if there is a way to author short URLs by putting the API Key in the url itself (i.e. not use headers). This is necessary for me as I use dlvr.it to post links to various news articles, and it supports 3rd party shortening, but _only_ via a single URL specifier so I can't add my API key. My previous hosted shortener - Yourls - allowed for this but I like shlink's UI and some of the additional features so am hoping I can use it here as well

![image](https://github.com/shlinkio/shlink/assets/1967433/f52d24aa-32c7-49a2-a903-4d8376f9724b)

Looking thru the docs it doesn't seem like I can do this, but thought I'd ask.

Thanks!

**Comment by acelaya** [comment, marked as answer]
Yes, that feature was implemented: https://api-spec.shlink.io/#/Short%20URLs/shortenUrl

The endpoint is marked as deprecated basically because ~using that is a terrible idea from a security point of view, and I don't want people to end up using it if they can avoid it.~ I want to keep a single way to do things that follows proper API design, and that GET endpoint won't receive more features.

~It's very unfortunate that services that allow you to integrate your own URL shortener are so limited, so I would suggest trying to contact them and letting them know how bad that is (your API key will travel in plain text, even for encrypted connections).~

~That said, the endpoint won't go away, so you can use it if there's no alternative, but stop using it as soon as you can 😅~

EDIT: see my clarification below.

**Reply by acelaya** [reply]
> It's very unfortunate that services that allow you to integrate your own URL shortener are so limited, so I would suggest trying to contact them and letting them know how bad that is (your API key will travel in plain text, even for encrypted connections).
>
> That said, the endpoint won't go away, so you can use it if there's no alternative, but stop using it as soon as you can 😅

Sorry, this is actually not true. The query string is in fact encrypted when using HTTPS, but still, it's a bit annoying they don't support more options and a dedicated endpoint is required.

**Reply by bc3tech** [reply]
Ha! my eyes skipped right over that striked-out API, thanks!
